graph/api/src/main/scala/org/apache/spark/graph/api/GraphElementFrame.scala

```diff
@@ -0,0 +1,260 @@
+   * @param idColumn  column that contains the node identifier
+   * @param labelSet  labels that are assigned to all nodes
+   * @since 3.0.0
+   */
+  def create(df: DataFrame, idColumn: String, labelSet: Set[String]): NodeFrame = {
+    val properties = (df.columns.toSet - idColumn)
+      .map(columnName => columnName -> columnName)
+      .toMap
+    create(df, idColumn, labelSet, properties)
+  }
+
+  /**
+   * Describes how to map an initial DataFrame to nodes.
+   *
+   * All columns apart from the given `idColumn` are mapped to node properties.
+   *
+   * @param df        DataFrame containing a single node in each row
+   * @param idColumn  column that contains the node identifier
+   * @param labelSet  labels that are assigned to all nodes
+   * @param properties mapping from property keys to corresponding columns
+   * @since 3.0.0
+   */
+  def create(
+      df: DataFrame,
+      idColumn: String,
+      labelSet: Set[String],
+      properties: Map[String, String]): NodeFrame = {
+    NodeFrame(df, idColumn, labelSet, properties)
+  }
+
+  /**
+   * Describes how to map an initial DataFrame to nodes.
+   *
+   * All columns apart from the given `idColumn` are mapped to node properties.
+   *
+   * @param df        DataFrame containing a single node in each row
+   * @param idColumn  column that contains the node identifier
+   * @param labelSet  labels that are assigned to all nodes
+   * @since 3.0.0
+   */
+  def create(df: DataFrame, idColumn: String, labelSet: java.util.Set[String]): NodeFrame = {
+    create(df, idColumn, labelSet.asScala.toSet)
+  }
+
+  /**
+   * Describes how to map an initial DataFrame to nodes.
+   *
+   * All columns apart from the given `idColumn` are mapped to node properties.
+   *
+   * @param df        DataFrame containing a single node in each row
+   * @param idColumn  column that contains the node identifier
+   * @param labelSet  labels that are assigned to all nodes
+   * @param properties mapping from property keys to corresponding columns
+   * @since 3.0.0
+   */
+  def create(
+      df: DataFrame,
+      idColumn: String,
+      labelSet: java.util.Set[String],
+      properties: java.util.Map[String, String]): NodeFrame = {
+    val scalaLabelSet = labelSet.asScala.toSet
+    val scalaProperties = properties.asScala.toMap
+    NodeFrame(df, idColumn, scalaLabelSet, scalaProperties)
+  }
+
+}
+
+/**
+ * Describes how to map a DataFrame to nodes.
+ *
+ * Each row in the DataFrame represents a node which has exactly the labels defined by the given
+ * label set.
+ *
+ * @param df         DataFrame containing a single node in each row
+ * @param idColumn   column that contains the node identifier
+ * @param labelSet   labels that are assigned to all nodes
+ * @param properties mapping from property keys to corresponding columns
+ * @since 3.0.0
+ */
+case class NodeFrame private[graph] (
+    df: DataFrame,
+    idColumn: String,
```

Comment:
When do we verify the `idColumn` column exists?

Reply:
Are we supposed to check that? Catalyst logical planning will fail if we try to access a column that is not there and throw a descriptive error message. If we want to fail early, we can add this to the companion objects of NodeFrame / RelationshipFrame. wdyt?